src/NuGet.Services.BasicSearch/Startup.cs

```diff
@@ -172,19 +173,22 @@ private void ReopenCallback(object state)
             }
         }
 
-        private bool InitializeSearcherManager(IConfiguration configuration, Directory directory, ILoader loader, ILoggerFactory loggerFactory)
+        private async Task<bool> InitializeSearcherManager(ISettingsProvider settings, Directory directory, ILoader loader, ILoggerFactory loggerFactory)
         {
             const int maxRetries = 10;
 
             try
             {
-                Retry.Incremental(
-                    () =>
+                await Retry.IncrementalAsync(
+                    async () =>
                     {
                         var stopwatch = Stopwatch.StartNew();
 
-                        _searcherManager = NuGetSearcherManager.Create(configuration, loggerFactory, directory, loader);
-                        _searcherManager.Open();
+                        var searcherManager = await NuGetSearcherManager.Create(settings, loggerFactory, directory, loader);
+                        await _searcherManager.OpenAsync();
+
+                        // Guarantee that the searcher manager has been opened before we set it.
```

Comment:
It's probably not, but I did it because previously the two lines above ran synchronously, but now this is an asynchronous context. I want to make the operations above atomic to reduce the likelihood of a threading problem, so I moved the assignment of `_searcherManager`. Not sure if it's necessary.